flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -31,6 +31,10 @@
 	/**
 	 * Asynchronously register a partition and its producer with the shuffle service.
 	 *
+	 * <p>IMPORTANT: the returned future must be completed due to limitations in the default scheduler.
+	 * The default scheduler only supports synchronous registration at the moment.
```

Comment:
This line and the one below effectively do not provide any new information; I'd remove them.